### DISCUSSION TITLE
Precision issues?

### DISCUSSION THREAD

**Discussion by tanevitch** [AI-team-UoA/pyJedAI#8] [Q&A] [answered]
Hi there! I executed a script to compare all building blocks methods with all comparison cleaning methods applied to a dirty ER approach. The results show a low precision, as you can see en the picture.
So I was wondering... could a previous step be necessary, like training the model or maybe the ground truth file should cointain a certain amount of positive and negative deduplicates? Do you have any idea about why it's scoring low?
Thank you, have a nice week
![Figure_1](https://user-images.githubusercontent.com/66225963/235375089-d41e42e9-a86e-4da5-9b4d-06437cf0fce0.png)

**Comment by gpapadis** [comment, marked as answer]
Hi, thanks for reaching out!  
I apologize for the late response. I just came accross your question.
Low precision is normal for blocking workflows. Their goal is to achieve high recall, which I see they do, while reducing the computational cost when compared to the brute force approach (i.e., all pairs of input entities). The candidate pairs generated by blocking are then processed by matching methods, which indeed raise precision at a low cost in recall.
